ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -358,5 +375,32 @@ public override void OnReactInstanceDispose()
 
             return null;
         }
+
+        private async Task<bool> IsCancelledOrDisposed()
+        {
+            try
+            {
+                var token = _cancelSource.Token;
```

Comment:
Can we use the mutex and check the `IsCanceled` property rather than relying on exception behavior? I know we'll likely need the catch blocks anyway, but it might prevent a few frequent exceptions.